src/views/collections/CollectionView/CollectionView.vue

```diff
@@ -0,0 +1,151 @@
+    </div>
+    <Comments :comments="comments" :nid="collection.nid" />
+
+    <template #sidebar="{ isInSidebar }">
+      <SidebarPanel
+        :isInSidebar="isInSidebar"
+        :header="$t('collection-info:side-bar-header')"
+        headerIcon="@/assets/info.svg"
+      >
+        <template #header-icon>
+          <img src="@/assets/info.svg" />
+        </template>
+        <ul style="padding: 0; list-style-type:none" v-if="collection">
+          <li>
+            <img :src="collection.userpicture" class="icon" />{{ collection.username }}
+          </li>
+          <li>
+            <img src="@/assets/calendar.svg" class="icon" />{{ collection.created }}
+          </li>
+          <div v-if="editRights">
+            <li>  
+              <b-button
+              type="submit"
+              variant="primary"
+              class="submit-button"
+              :href="`/collections/${nid}/edit`"
+              >
+                {{ "Edit" }}
+              </b-button>
+            </li>
+          </div>
+        </ul>
+      </SidebarPanel>
+    </template>
+  </EternaPage>
+  <Preloader v-else style="margin-top: 10rem;" />
+</template>
+
+<script lang="ts">
+  import { Component, Vue, Mixins } from 'vue-property-decorator';
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import PuzzleCard from '@/components/Cards/PuzzleCard.vue';
+  import CollectionCard from '@/components/Cards/QuestCard.vue';
+  import SidebarPanel from '@/components/Sidebar/SidebarPanel.vue';
+  import Preloader from '@/components/PageLayout/Preloader.vue';
+  import { PuzzleList, PuzzleItem, ClearedPuzzle, RoadmapAchievement, CollectionItem, CollectionResponse, CommentItem } from '@/types/common-types';
+  import FetchMixin from '@/mixins/FetchMixin';
+  import Comments from '@/components/PageLayout/Comments.vue';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      PuzzleCard,
+      CollectionCard,
+      SidebarPanel,
+      Preloader,
+      Comments,
+    },
+  })
+  export default class CollectionView extends Mixins(FetchMixin) {
+    puzzles: PuzzleItem[] = [];
+
+    cleared: ClearedPuzzle[] = [];
+
+    collection: CollectionItem | null = null;
+
+    comments: CommentItem[] = [];
+
+    editRights: Boolean = false;
+
+    nid: string|undefined;
+
+    async fetch(){
+      const res = (
+        await this.$http.get(`/get/?type=collection&nid=${this.$route.params.id}`)
+      ).data.data as CollectionResponse;
+      this.collection = res.collection;
+      this.comments = res.comments;
+      this.nid = res.collection.nid;
+      if(this.collection.username === this.$vxm.user.username) this.editRights = true;
+      const puzzlelist = this.collection.puzzles.split(",");
+      Object.values(puzzlelist).forEach(async puzz => this.puzzles.push((await this.$http.get(`/get/?type=puzzle&nid=${parseInt(puzz, 10)}`)).data.data as PuzzleItem));
```

Comment:
As implemented, I think this fires off all the async calls in parallel, and then as each comes in adds them to the list. Is this the desired behavior? Or would it be better to use `.map` to turn puzzlelist into a list of promises so that you can `await Promise.all` on that before adding to the list? Or, as I think I mentioned before, could the puzzles details be returned automatically with the collection API call by the backend?

Edit: I think this was changed on the edit page, but not here